client.go

```diff
@@ -29,41 +38,52 @@ type Config struct {
 // NewClient - returns new redis client with provided config options
 func NewClient(c Config) (*Client, error) {
 	if c.Addr == "" {
-		return nil, errors.New("address is missing")
+		return nil, ErrEmptyAddress
 	}
 
 	if c.Password == "" {
-		return nil, errors.New("password is missing")
+		return nil, ErrEmptyPassword
+	}
+
+	if c.TTL == 0 {
+		return nil, ErrInvalidTTL
 	}
 
 	return &Client{
-		client: goredis.NewClient(&goredis.Options{
-			Addr:     c.Addr,
+		client:  RedisClient{client:redis.NewClient(&redis.Options{
+			Addr: c.Addr,
 			Password: c.Password,
-			DB:       c.Database,
-		}),
-		ttl: c.TTL,
+			DB: c.Database,
+		})},
+		ttl:     c.TTL,
 	}, nil
 }
 
-//Set add session to redis
+// Set - add session to redis
 func (c *Client) Set(s *session.Session) error {
 	if s == nil {
-		log.Event(ctx, "session is empty", log.ERROR)
-		return errors.New("session is empty")
+		return ErrEmptySession
 	}
 
 	json, err := s.MarshalJSON()
 	if err != nil {
-		log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
 		return err
 	}
 
-	msg, err := c.client.Set(s.ID, json, c.ttl).Result()
+	err = c.client.Set(s.ID, string(json), c.ttl).Err()
 	if err != nil {
-		log.Event(ctx, msg, log.Error(err), log.ERROR)
+		return err
```

Comment:
This will be an error from within the redis client I would considering wrapping a new error around this to give _our_ context _i.e redis client.set returned an unexpected errror_

(see https://blog.golang.org/go1.13-errors Wrapping errors with %w)